xls/ir/verify_node.cc

```diff
@@ -1465,12 +1465,41 @@ class NodeChecker : public DfsVisitor {
     return absl::OkStatus();
   }
 
-  // Verifies all operands are BitsType with the same bit count.
-  absl::Status ExpectOperandsSameBitsType(Node* node) const {
+  // Verifies all operands are BitsType with the same bit count as the first
+  // operand.
+  absl::Status ExpectOperandsAreBitsAndSameWidth(Node* node) const {
     if (node->operand_count() == 0) {
       return absl::OkStatus();
     }
+    XLS_RETURN_IF_ERROR(ExpectOperandHasBitsType(node, 0));
     Type* type = node->operand(0)->GetType();
+    // Already checked operand 0 is BitsType. Now check other operands.
+    for (int64_t i = 1; i < node->operand_count(); ++i) {
+      XLS_RETURN_IF_ERROR(ExpectOperandHasBitsType(node, i));
+      if (node->operand(i)->GetType()->AsBitsOrDie()->bit_count() !=
```

Comment:
nit: couldn't you use `node->operand(i)->BitCountOrDie()` ?

Reply:
I have a slight preference for this way, since it's is showing you the type is indeed BitsType after the call to `ExpectOperandHasBitsType` succeeds. Tbh it's kind of weird BitCountOrDie assumes the node has a BitsType output, looking at it we might want to remove that since it's not really a sound polymorphic assumption.